pkg/runtime/profiler.go

```diff
@@ -0,0 +1,135 @@
+package runtime
+
+import (
+	"fmt"
+	"os"
+	goruntime "runtime"
+	"runtime/pprof"
+	"runtime/trace"
+)
+
+// ProfilerConfig defines the configuration of a Profiler
+type ProfilerConfig struct {
+	CPUProfile         bool
+	CPUProfileFileName string
+	MemProfile         bool
+	MemProfileRate     int
+	MemProfileFileName string
+	Trace              bool
+	TraceFileName      string
+}
+
+// Profiler defines the methods to control execution profiling
+type Profiler interface {
+	// Start initiates the tracing. If already active, has no effect
+	Start() error
+	// Stop terminates the tracing. If not started, has no effect
+	Stop() error
+}
```

Comment:
Do we need an interface for `Profiler`? From the looks of it it seems that the zero-value of the implementation is already a noop in case we need one.

Reply:
I don't understand the comment. 
If we don't have an interface, how do we control it? And. What do you mean by a "noop".

Can you please elaborate?

Reply:
> We can export the implementation and use it directly.

How do we mock it? Please take into consideration that the whole purpose of this PR is to allow mocking the interfaces we are introducing.

Reply:
> I mean that the zero-value does nothing if you call Start or Stop on it, which is great because you don't need a fake in case you need it

Still, I don't understand. The interface is there so we can mock it and validate if it was called by the command when we expected it to.